[2024-09-21T16:05:44.419+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-09-21T16:05:44.444+0000] {taskinstance.py:2603} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: employee_data_marts_dag.create_mart_performance manual__2024-09-21T16:04:43.853129+00:00 [queued]>
[2024-09-21T16:05:44.463+0000] {taskinstance.py:2603} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: employee_data_marts_dag.create_mart_performance manual__2024-09-21T16:04:43.853129+00:00 [queued]>
[2024-09-21T16:05:44.464+0000] {taskinstance.py:2856} INFO - Starting attempt 1 of 1
[2024-09-21T16:05:44.485+0000] {taskinstance.py:2879} INFO - Executing <Task(PythonOperator): create_mart_performance> on 2024-09-21 16:04:43.853129+00:00
[2024-09-21T16:05:44.495+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=4906) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-09-21T16:05:44.497+0000] {standard_task_runner.py:72} INFO - Started process 4907 to run task
[2024-09-21T16:05:44.495+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'employee_data_marts_dag', 'create_mart_performance', 'manual__2024-09-21T16:04:43.853129+00:00', '--job-id', '54', '--raw', '--subdir', 'DAGS_FOLDER/employee_data_marts_dag_update.py', '--cfg-path', '/tmp/tmpff3ai52f']
[2024-09-21T16:05:44.498+0000] {standard_task_runner.py:105} INFO - Job 54: Subtask create_mart_performance
[2024-09-21T16:05:44.567+0000] {task_command.py:467} INFO - Running <TaskInstance: employee_data_marts_dag.create_mart_performance manual__2024-09-21T16:04:43.853129+00:00 [running]> on host 942f0f86d762
[2024-09-21T16:05:44.680+0000] {taskinstance.py:3122} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='employee_data_marts_dag' AIRFLOW_CTX_TASK_ID='create_mart_performance' AIRFLOW_CTX_EXECUTION_DATE='2024-09-21T16:04:43.853129+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-09-21T16:04:43.853129+00:00'
[2024-09-21T16:05:44.682+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-09-21T16:05:56.456+0000] {taskinstance.py:3301} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/employee_data_marts_dag_update.py", line 94, in create_mart_performance
    mart_performance = spark.sql("""
                       ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/session.py", line 1631, in sql
    return DataFrame(self._jsparkSession.sql(sqlQuery, litArgs), self)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/py4j/java_gateway.py", line 1322, in __call__
    return_value = get_return_value(
                   ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/errors/exceptions/captured.py", line 185, in deco
    raise converted from None
pyspark.errors.exceptions.captured.AnalysisException: [UNRESOLVED_COLUMN.WITH_SUGGESTION] A column or function parameter with name `PerformanceScore` cannot be resolved. Did you mean one of the following? [`PerformanceID`, `Comments`, `Name`, `ReviewPeriod`, `EmployeeID`].; line 5 pos 12;
'Project [EmployeeID#1, Name#2, 'PerformanceScore, 'ReviewDate]
+- SubqueryAlias fact_performance
   +- View (`fact_performance`, [PerformanceID#0,EmployeeID#1,Name#2,ReviewPeriod#3,Rating#4,Comments#5])
      +- Relation [PerformanceID#0,EmployeeID#1,Name#2,ReviewPeriod#3,Rating#4,Comments#5] JDBCRelation(warehouse.fact_performance) [numPartitions=1]

[2024-09-21T16:05:56.489+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=employee_data_marts_dag, task_id=create_mart_performance, run_id=manual__2024-09-21T16:04:43.853129+00:00, execution_date=20240921T160443, start_date=20240921T160544, end_date=20240921T160556
[2024-09-21T16:05:56.518+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-09-21T16:05:56.520+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 54 for task create_mart_performance ([UNRESOLVED_COLUMN.WITH_SUGGESTION] A column or function parameter with name `PerformanceScore` cannot be resolved. Did you mean one of the following? [`PerformanceID`, `Comments`, `Name`, `ReviewPeriod`, `EmployeeID`].; line 5 pos 12;
'Project [EmployeeID#1, Name#2, 'PerformanceScore, 'ReviewDate]
+- SubqueryAlias fact_performance
   +- View (`fact_performance`, [PerformanceID#0,EmployeeID#1,Name#2,ReviewPeriod#3,Rating#4,Comments#5])
      +- Relation [PerformanceID#0,EmployeeID#1,Name#2,ReviewPeriod#3,Rating#4,Comments#5] JDBCRelation(warehouse.fact_performance) [numPartitions=1]
; 4907)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 2995, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3149, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3173, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/employee_data_marts_dag_update.py", line 94, in create_mart_performance
    mart_performance = spark.sql("""
                       ^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/session.py", line 1631, in sql
    return DataFrame(self._jsparkSession.sql(sqlQuery, litArgs), self)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/py4j/java_gateway.py", line 1322, in __call__
    return_value = get_return_value(
                   ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/errors/exceptions/captured.py", line 185, in deco
    raise converted from None
pyspark.errors.exceptions.captured.AnalysisException: [UNRESOLVED_COLUMN.WITH_SUGGESTION] A column or function parameter with name `PerformanceScore` cannot be resolved. Did you mean one of the following? [`PerformanceID`, `Comments`, `Name`, `ReviewPeriod`, `EmployeeID`].; line 5 pos 12;
'Project [EmployeeID#1, Name#2, 'PerformanceScore, 'ReviewDate]
+- SubqueryAlias fact_performance
   +- View (`fact_performance`, [PerformanceID#0,EmployeeID#1,Name#2,ReviewPeriod#3,Rating#4,Comments#5])
      +- Relation [PerformanceID#0,EmployeeID#1,Name#2,ReviewPeriod#3,Rating#4,Comments#5] JDBCRelation(warehouse.fact_performance) [numPartitions=1]

[2024-09-21T16:05:56.583+0000] {local_task_job_runner.py:261} INFO - Task exited with return code 1
[2024-09-21T16:05:56.610+0000] {taskinstance.py:3891} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-09-21T16:05:56.614+0000] {local_task_job_runner.py:240} INFO - ::endgroup::
